src/Infrastructure/LeanCode.ForceUpdate/LeanCode.ForceUpdate.Services/VersionHandler.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.AspNetCore.Http;
+
+namespace LeanCode.ForceUpdate.Services;
+
+public class VersionHandler
+{
+    private readonly AndroidVersionsConfiguration androidConfiguration;
+    private readonly IOSVersionsConfiguration iOSConfiguration;
+
+    public VersionHandler(AndroidVersionsConfiguration androidConfiguration, IOSVersionsConfiguration iOSConfiguration)
+    {
+        this.androidConfiguration = androidConfiguration;
+        this.iOSConfiguration = iOSConfiguration;
+    }
+
+    public virtual Task<VersionSupportResult> CheckVersionAsync(Version version, Platform platform, HttpContext context)
```

Comment:
Why is this `async`?

Reply:
As @Saancreed said above:
> Actually, I can probably think of some cases where overrides of this method would want to perform some async call (e.g. we want to force update for specific tenants only and we store that information in the database). Can we make it `virtual Task<VersionSupport> CheckVersionAsync` instead?
> 
> (Aaand that makes my previous suggestion about refactoring `Task.FromResult` no longer valid, sorry sweat_smile)

Reply:
@Saancreed what kind of usecases you had in mind? I can't find one :(

Reply:
> we want to force update for specific tenants only and we store that information in the database

Like, we have a new feature that's not yet available for everyone which needs a new version of the application to work correctly, so we'd need to force an update only for some users that actually have that feature enabled and there's no need for everyone else to update just yet.

Yes, this is probably not something that will come up very often but restricting this to just sync calls feels a bit shortsighted to me. Even having this as `ValueTask` that usually completes synchronously would be preferable to reinventing the wheel once an actual usecase is found.

Reply:
I fear that this is not _the_ feature for this. The package is meant to be super-simple, without many features (esp. that I think more advanced features would need _more_ data in the query).

Nevertheless, I agree - we should not limit ourselves.

But let's move to `ValueTask` @wojtek2288 :P